app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala

```diff
@@ -216,41 +216,67 @@ class RelationshipsController @Inject()(
       }
   }
 
-  private def validateParams(serviceKey: String, clientType: String, clientId: String): Either[String, EnrolmentKey] =
+  private def validateParams(serviceKey: String, clientType: String, clientId: String)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[Either[String, EnrolmentKey]] =
     (serviceKey, clientType) match {
       // "special" cases
-      case ("IR-SA", "ni" | "NI") if Nino.isValid(clientId) => Right(EnrolmentKey("IR-SA", Nino(clientId)))
+      case ("IR-SA", "ni" | "NI") if Nino.isValid(clientId) =>
+        Future.successful(Right(EnrolmentKey("IR-SA", Nino(clientId))))
       case (Service.MtdIt.id, "ni" | "NI") if Nino.isValid(clientId) =>
-        Right(EnrolmentKey(Service.MtdIt.id, Nino(clientId)))
-      case ("HMCE-VATDEC-ORG", "vrn") if Vrn.isValid(clientId) => Right(EnrolmentKey("HMCE-VATDEC-ORG", Vrn(clientId)))
+        Future.successful(Right(EnrolmentKey(Service.MtdIt.id, Nino(clientId))))
+      case ("HMCE-VATDEC-ORG", "vrn") if Vrn.isValid(clientId) =>
+        Future.successful(Right(EnrolmentKey("HMCE-VATDEC-ORG", Vrn(clientId))))
+      case (Service.Cbc.id, CbcIdType.enrolmentId) => // need to fetch the UTR to make a 'complete' enrolment key
+        esConnector.findUtrForCbcId(CbcId(clientId)).map {
+          case Some(utr) =>
+            Right(
+              EnrolmentKey(
+                Service.Cbc.id,
+                Seq(Identifier(CbcIdType.enrolmentId, clientId), Identifier("UTR" /* Not "SAUTR"! */, utr.value))))
```

Comment:
@geoffreywatson the order here may be important?

https://github.com/hmrc/service-enrolment-config/blob/main/conf/service_identifiers_json/prod/HMRC-CBC-ORG.json

In the service-enrolment-config, UTR is first, which is why in the stubs it's generated that way around. Also we'll have to account for this in other places if we need the full enrolment key. Do we know if it could ever be reversed?